requests_ntlm/requests_ntlm.py

```diff
@@ -86,22 +88,23 @@ def retry_using_http_NTLM_auth(self, auth_header_field, auth_header,
         # get the challenge
         auth_header_value = response2.headers[auth_header_field]
 
-        ntlm_header_value = [
-            s for s in (val.lstrip() for val in auth_header_value.split(','))
-            if s.startswith('NTLM ')
-        ][0].strip()
+        auth_strip = auth_type + ' '
+
+        ntlm_header_value = list(filter(
+            lambda s: s.startswith(auth_strip), auth_header_value.split(',')
+        ))[0].strip()
```

Comment:
Rather than create the intervening list here, it might be better to use `next()` to return the first element.